### DISCUSSION TITLE
Pre handler hook with prefix route

### DISCUSSION THREAD

**Discussion by rndy28** [fastify/fastify#5025] [Q&A] [answered]
Hi there, I am new on fastify

Is there any way I can attach prehandler to specific prefix like express router?

example express router

```ts
// users.route.ts
const router = Router();

router.get("/", userController.getUsers);
router.put("/:id", userController.editUser);

// main.ts
app.use("/api/users", requireUser, usersRoutes);
```

here's what I did so far,  I thought that prehandler is executing only on this prefix
```ts
  app.register(
    (instance, _, done) => {
      instance.get("/", getUsers);
      instance.put("/:id", () => {});
      instance.delete("/:id", () => {});
      done();
    },
    { prefix: "/users" }
  ).addHook("preHandler", isAuthed);
```
But the pre handler is executing on all of the routes

**Comment by Eomm** [comment, marked as answer]
Write instance.addHook instead.
Your code applies the hook to the app instance 

There is another technique using the onroute hook. I call it aspect oriented programming in my fastify book 👍

**Reply by rndy28** [reply]
Hi thanks for the answer.  Owh thats why its apply to the whole routes (my dumbass think its already different thing since I specify the prefix 🤣), Its working now, thanks 🙂
